src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java

```diff
@@ -194,7 +197,12 @@ private String getProjectedFields(String b, Class resultClass, String[] fields)
 
 		String projectedFields = i(b) + ".*";
 		if (resultClass != null) {
-			PersistentEntity persistentEntity = couchbaseConverter.getMappingContext().getPersistentEntity(resultClass);
+			PersistentEntity persistentEntity;
+			if (ClassUtils.isPrimitiveOrWrapper(resultClass)) {
+				persistentEntity = new BasicCouchbasePersistentEntity(ClassTypeInformation.from(resultClass));
```

Comment:
Similar issue as with `N1qlQueryCreator`. The entity should be determined on the query method level and if the return type qualifies as entity, the entity to be used should be refined into the type instead of creating `PersistentEntity` instances.